### ISSUE TITLE
three data.table merge behavior inconsistency

### ISSUE THREAD

**Issue by ajdamico** [Rdatatable/data.table#4772]
i looked around for related discussions and didn't find anything specific to this?  i posted on SO and the accepted answerer there provided more detail about how it's likely a bug and also an easy fix--  https://stackoverflow.com/questions/64464386/three-data-table-merge-behavior-inconsistency/64464931#64464931

thanks very very much for this excellent software!

	library(data.table)

	x <- data.table( a = 1:3 )
	y <- data.table( a = 2:4 )
	z <- data.table( a = 3:5 )

	# works
	merge( x , y )
	# works
	merge( y , z )

	# fails
	merge( x , merge( y , z ) )
	# Error in merge.data.table(x, merge(y, z)) :
	#   A non-empty vector of column names for `by` is required.

	# works
	merge( merge( x , y ) , z )

**Comment by ecoRoland2** [comment]
More minimal example:

```R
library(data.table)
x <- data.table(a = 1:3)
y <- data.table(a = 2:4, key = "a")

merge(x, y)
#Error in merge.data.table(x, y) : 
#  A non-empty vector of column names for `by` is required.
```

I believe you can simply change lines 35 and 37 of merge.R to `if (length(by) == 0L)`.

**Comment by ecoRoland2** [comment]
@ajdamico Please change the title to "merge fails with intersecting names if y is keyed"

**Comment by MichaelChirico** [comment]
For ease of future reference: this was actually closed by #5183.

**Comment by ajdamico** [comment]
thank you!!
